File: Server/src/main/java/fr/makizart/database/table/Project.java
package fr.makizart.database.table;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Entity
@Table(name = "PROJECT")
public class Project {

    public Project() {
        arResource = Objects.requireNonNullElse(this.arResource, new ArrayList<>());
    }

    @Id
    @GeneratedValue
    private Long id;

    @OneToMany(cascade = CascadeType.ALL)
    private List<ArResource> arResource;

    @Column(name="NAME", nullable = false)
    private String name;

    public List<ArResource> getArResource() {
        return arResource;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }


}

File: WebView/site/src/app/commons/Project.ts
export class Project{
  private _id : number;
  private _name : string;
  private _creationDate : string;

  constructor(id: number, name: string, creationDate: string) {
    this._id = id;
    this._name = name;
    this._creationDate = creationDate;
  }

  set name(value: string) {
    this._name = value;
  }

  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get creationDate(): string {
    return this._creationDate;
  }
}
